// @azure/arm-botservice/esm/operations/hostSettings.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is
 * regenerated.
 */
import * as msRest from "@azure/ms-rest-js";
import * as Mappers from "../models/hostSettingsMappers";
import * as Parameters from "../models/parameters";
/** Class representing a HostSettings. */
var HostSettings = /** @class */ (function () {
    /**
     * Create a HostSettings.
     * @param {AzureBotServiceContext} client Reference to the service client.
     */
    function HostSettings(client) {
        this.client = client;
    }
    HostSettings.prototype.get = function (options, callback) {
        return this.client.sendOperationRequest({
            options: options
        }, getOperationSpec, callback);
    };
    return HostSettings;
}());
export { HostSettings };
// Operation Specifications
var serializer = new msRest.Serializer(Mappers);
var getOperationSpec = {
    httpMethod: "GET",
    path: "subscriptions/{subscriptionId}/providers/Microsoft.BotService/hostSettings",
    urlParameters: [
        Parameters.subscriptionId
    ],
    queryParameters: [
        Parameters.apiVersion
    ],
    headerParameters: [
        Parameters.acceptLanguage
    ],
    responses: {
        200: {
            bodyMapper: Mappers.HostSettingsResponse
        },
        default: {
            bodyMapper: Mappers.ErrorModel
        }
    },
    serializer: serializer
};
